spec/ics-2-consensus-requirements/README.md

```diff
@@ -0,0 +1,76 @@
+
+Requirements:
+
+1. Headers have only one direct child
+If a blockchain has deterministic safety(as opposed to probabilistic safety in Nakamoto consensus), then there cannot be more than one child for each block. In Tendermint, this assumption breaks when +1/3 validators are malicious, producing multiple blocks those are direct children of a single block and all of them are verifiable with the parent. It makes conflicting packets delivered from the failed chain, so for the chains who are receiving from this chain, **fraud proof #10** mechanism should be applied in this case. However, he failed chain itself also need to be recovered and reconnected again. It will be covered in **byzantine recovery strategies #6**.
+
+2. Headers have at least one direct child
+There is at least one direct child for all blocks, meaning that the lightclient logic can proceed the blocks one by one even in the worst case. If it not satisfied then there can be a point where the lightclient stops and cannot proceed, which halts IBC connection unexpectedly. This also can be violated when the blocks are restarted out of the consensus, for example in Tendermint, (1/2 < validators < 2/3) forked out the chain. This also will be covered in **byzantine recovery strategies**.
+
+3. If a block verifies another block then it is a descendant of the block
+Lightclient should not verify packets which are not in its chain. If the verifier returns true for a block that is not a descendant, it simply means that there is an error in the lightclient logic.
+
+4. Header can have a state only if it is a valid transition from the parent's(see the next paragraph)
+Header cannot have a state which is not a transition from its parent. This requires that the chain satisfies the IBC protocol and runs it honestly. In case where the validators become malicious and send invalid messages, the application logic on the destination chain should prevent further interchain infection.
+ 
+These requirements allow channels to work safely without concerning about double spending attack. 
+If a block is submitted to the chain(and optionally passed some challenge period for fraud proof), then it is assured that the packet is finalized so the application logic can process it.
+
+### Technical Specification
+
+In a real world, `Header`s do not have to contain the full state data; it can contain an aggregated data set identifier, for example a merkle root, where the actual data latter given can be proven that it is a member of that set. If we don't consider about the data availability problem, this partial blocks, which is be called `Header`s, still satisfies `Header`.
+
+Following functions exist over `Header`s.
+
+* `height : Header -> int`
+Returns the height of the block.
+
+* `verify : Header -> Header -> Maybe<LightClientProof> -> Maybe<error>`
+Verifies the latter block is a descendant of the former block, as defined in **lightclient specification**. Can take an additional lightclient proof argument.
```

Comment:
Is **lightclient specification** referring to https://github.com/cosmos/ics/issues/13 ? 

We should make the dependency clear up front somewhere

Reply:
I made a comment in #25: https://github.com/cosmos/ics/issues/25#issuecomment-476727693 will reflect that on the spec file

Reply:
I think this function should instead return a new `RootOfTrust`, or we should have an extra function for that purpose.

Reply:
Oh I guess you mean `RootOfTrust` when you say `Header` - we should switch those, `Header` is the `LightClientProof` consensus transcript fragment type.

Reply:
Maybe `verifyAndUpdate : RootOfTrust -> Header -> Either Error RootOfTrust`

Reply:
I like this statement to update trust